Framework/Algorithms/src/MaskNonOverlappingBins.cpp

```diff
@@ -0,0 +1,136 @@
+#include "MantidAlgorithms/MaskNonOverlappingBins.h"
+
+#include "MantidAPI/MatrixWorkspace.h"
+#include "MantidKernel/MultiThreaded.h"
+
+namespace {
+/// Constants for the algorithm's property names.
+namespace Prop {
+std::string const COMPARISON_WS{"ComparisonWorkspace"};
+std::string const INPUT_WS{"InputWorkspace"};
+std::string const MASK_PARTIAL{"MaskPartiallyOverlapping"};
+std::string const OUTPUT_WS{"OutputWorkspace"};
+} // namespace Prop
+} // namespace
+
+namespace Mantid {
+namespace Algorithms {
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+
+DECLARE_ALGORITHM(MaskNonOverlappingBins)
+
+/// Algorithms name for identification. @see Algorithm::name
+std::string const MaskNonOverlappingBins::name() const {
+  return "MaskNonOverlappingBins";
+}
+
+/// Algorithm's version for identification. @see Algorithm::version
+int MaskNonOverlappingBins::version() const { return 1; }
+
+/// Algorithm's category for identification. @see Algorithm::category
+std::string const MaskNonOverlappingBins::category() const {
+  return "Transforms\\Masking";
+}
+
+/// Algorithm's summary for use in the GUI and help. @see Algorithm::summary
+std::string const MaskNonOverlappingBins::summary() const {
+  return "Marks bins in InputWorkspace which are out of the X range of the "
+         "second workspace.";
+}
+
+/// Return a list of the names of related algorithms.
+std::vector<std::string> const MaskNonOverlappingBins::seeAlso() const {
+  return {"MaskBins", "MaskBinsIf"};
+}
+
+/** Initialize the algorithm's properties.
+ */
+void MaskNonOverlappingBins::init() {
+  declareProperty(Kernel::make_unique<API::WorkspaceProperty<>>(
+                      Prop::INPUT_WS, "", Kernel::Direction::Input),
+                  "A workspace to mask.");
+  declareProperty(Kernel::make_unique<API::WorkspaceProperty<>>(
```

Comment:
I think that you can restrict these to be `WorkspaceProperty<API::MatrixWorkspace>` you don't want to allow Workspaces of other types.

Reply:
I thought the default typename was `API::MatrixWorkspace`, but maybe I'm mistaken. I'll check it.

Reply:
It could be, in which case nothing to do.

Reply:
I can confirm that the typename indeed defaults to `MatrixWorkspace`.